### PR TITLE
Return accurate size of addressing mode dsp/dsp+cnst instr descriptor

### PR DESCRIPTION
**Pull request by kunalspathak**
During spmi `asmdiffs` with cross compiled jit, I noticed we hit the following assert https://github.com/dotnet/runtime/blob/035821a2eda0b4d87382f1bb24c69d25390bf8f5/src/coreclr/jit/emit.cpp#L1270 

It happens because for `ID_OP_AMD` and `ID_OP_AMD_CNS` format, we create `instrDescCnsAmd` but then in `emitSizeOfInsDsc()`, when we try to calculate the size, we calculate the size of `instrDescCnsDsp` instead. They don't match for cross compilation scenario like running `superpmi asmdiffs` for cross target. E.g. While running with `clrjit_win_x86_x64.dll`, `sizeof(instrDescCnsAmd) == 40` but `sizeof(instrDescCnsDsp) == 32`.  The is because `instrDescCnsDsp` contains a field of type `target_ssize_t ` which is different for x64 vs. x86.

https://github.com/dotnet/runtime/blob/035821a2eda0b4d87382f1bb24c69d25390bf8f5/src/coreclr/jit/emit.h#L1401

The fix is to return the size of right struct for these formats.

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

**Comment by echesakovMSFT** [comment]
@kunalspathak Have you confirmed our hypothesis that size of the instructions descriptors are the same with crossgen (x86) as we discussed yesterday?

File: src/coreclr/jit/emitxarch.cpp [hunk 1]
```diff
@@ -7440,6 +7438,30 @@ size_t emitter::emitSizeOfInsDsc(instrDesc* id)
                     return sizeof(instrDesc);
                 }
             }
+        case ID_OP_AMD:
+        case ID_OP_AMD_CNS:
+            if (id->idIsLargeCns())
```
Comment (BruceForstall) [inline src/coreclr/jit/emitxarch.cpp:7443]: Based on `emitNewInstrAmdCns`, I would expect it to be:
```
            if (id->idIsLargeCns())
            {
                if (id->idIsLargeDsp())
                {
                    return sizeof(instrDescCnsAmd);
                }
                else
                {
                    return sizeof(instrDescCns);
                }
            }
            else
            {
                if (id->idIsLargeDsp())
                {
                    return sizeof(instrDescAmd);
                }
                else
                {
                    return sizeof(instrDesc);
                }
            }
```

**Review by BruceForstall — CHANGES_REQUESTED** [review]

File: src/coreclr/jit/emitxarch.cpp (hunk 1 above)
Comment (kunalspathak) [reply src/coreclr/jit/emitxarch.cpp:7443]: Actually yes.

**Comment by kunalspathak** [comment]
> @kunalspathak Have you confirmed our hypothesis that size of the instructions descriptors are the same with crossgen (x86) as we discussed yesterday?

Yes @echesakovMSFT - they are same. I also crossgen [stind_i1.ilproj](https://github.com/dotnet/runtime/blob/master/src/tests/JIT/IL_Conformance/Old/Conformance_Base/stind_i1.ilproj) project which reproed the issue and used `R2RDump.exe` to see if things work.

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**